src/ScaledPolynomialTransformFitter.cc

```diff
@@ -0,0 +1,465 @@
+// -*- LSST-C++ -*-
+
+/*
+ * LSST Data Management System
+ * Copyright 2016 LSST/AURA
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include <map>
+
+#include "Eigen/LU" // for determinant, even though it's a 2x2 that doesn't use actual LU implementation
+
+#include "boost/math/tools/minima.hpp"
+
+#include "lsst/meas/astrom/ScaledPolynomialTransformFitter.h"
+#include "lsst/meas/astrom/detail/polynomialUtils.h"
+#include "lsst/afw/table/aggregates.h"
+#include "lsst/afw/table/Match.h"
+#include "lsst/afw/math/LeastSquares.h"
+
+#define LSST_ScaledPolynomialTransformFitter_TEST_IN_PLACE 1
+
+namespace lsst { namespace meas { namespace astrom {
+
+// A singleton struct that manages the schema and keys for polynomial fitting catalogs.
+class ScaledPolynomialTransformFitter::Keys {
```

Comment:
It's a similar pattern, though here it has nothing to do with persistence.  I suppose I just have a different instinct when it comes to singletons; if something is logically a singleton (a class with only one or two distinct instances), I tend to make it a singleton in code.  I wonder if @pschella has an opinion on whether this is generally wise or generally foolish?